serialization_test.go

```diff
@@ -429,43 +446,76 @@ func TestInferAcceptHeaderFromType(t *testing.T) {
 	})
 }
 
-func TestParseAcceptHeader(t *testing.T) {
+func TestInferAcceptHeader(t *testing.T) {
 	t.Run("can parse text/plain", func(t *testing.T) {
-		accept := parseAcceptHeader("text/plain", "Hello World")
+		accept := inferAcceptHeader("text/plain", "Hello World")
 		require.Equal(t, "text/plain", accept)
 	})
 
 	t.Run("can parse text/html", func(t *testing.T) {
-		accept := parseAcceptHeader("text/html", "<h1>Hello World</h1>")
+		accept := inferAcceptHeader("text/html", "<h1>Hello World</h1>")
 		require.Equal(t, "text/html", accept)
 	})
 
 	t.Run("can parse text/html from multiple options", func(t *testing.T) {
-		accept := parseAcceptHeader("text/html, text/plain", "<h1>Hello World</h1>")
+		accept := inferAcceptHeader("text/html", "<h1>Hello World</h1>")
 		require.Equal(t, "text/html", accept)
 	})
 
 	t.Run("can parse application/json", func(t *testing.T) {
-		accept := parseAcceptHeader("application/json", ans{})
+		accept := inferAcceptHeader("application/json", ans{})
 		require.Equal(t, "application/json", accept)
 	})
 
 	t.Run("can infer json", func(t *testing.T) {
-		accept := parseAcceptHeader("", response{})
+		accept := inferAcceptHeader("", response{})
 		require.Equal(t, "application/json", accept)
 	})
 
 	t.Run("can infer json", func(t *testing.T) {
-		accept := parseAcceptHeader("*/*", response{})
+		accept := inferAcceptHeader("*/*", response{})
 		require.Equal(t, "application/json", accept)
 	})
 
-	t.Run("can infer text/html from a real browser", func(t *testing.T) {
-		accept := parseAcceptHeader("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "<h1>Hello World</h1>")
+	t.Run("can infer text/plain", func(t *testing.T) {
+		accept := inferAcceptHeader("*/*", "Hello World")
+		require.Equal(t, "text/plain", accept)
+	})
+
+	t.Run("can infer text/html", func(t *testing.T) {
+		accept := inferAcceptHeader("*/*", HTML("Hello World"))
 		require.Equal(t, "text/html", accept)
 	})
 }
 
+func TestParseAcceptHeader(t *testing.T) {
+	t.Run("empty header", func(t *testing.T) {
+		accept := parseAcceptHeader(http.Header{})
+		require.Equal(t, []string{""}, accept)
+	})
+
+	t.Run("one value", func(t *testing.T) {
+		header := http.Header{}
+		header.Set("Accept", "text/html")
+		accept := parseAcceptHeader(header)
+		require.Equal(t, []string{"text/html"}, accept)
+	})
+
+	t.Run("two values", func(t *testing.T) {
+		header := http.Header{}
+		header.Set("Accept", "text/html,text/plain")
+		accept := parseAcceptHeader(header)
+		require.Equal(t, []string{"text/html", "text/plain"}, accept)
+	})
+
+	t.Run("can parse string from real browser", func(t *testing.T) {
+		header := http.Header{}
+		header.Set("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
+		accept := parseAcceptHeader(header)
+		require.Equal(t, []string{"text/html", "application/xhtml+xml", "application/xml;q=0.9", "*/*;q=0.8"}, accept)
```

Comment:
What can we do about the [Quality values](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values)?

I suggest to discard them, as the browser will always order the list of mime types correctly. We might do this in another PR if it really is needed.

The target array would be `require.Equal(t, []string{"text/html", "application/xhtml+xml", "application/xml", "*/*"}, accept)`

Reply:
👍 

Probably should switch the `*/*` to strings.Contains

Reply:
Ignoring Quality values entirely.